design-documents/graph-ql/path-validator.md

```diff
@@ -0,0 +1,100 @@
+# Magento Framework validators
+The purpose of this document is to a validator for relative url routes not only full urls commonly referred as absolute url routes.
+
+## Terminology
+`UrlValidator` - The existing framework validator `\Magento\Framework\Url\Validator`
+
+`RouteValidator` - The proposed `Magento\Framework\Url\RouteValidator`
```

Comment:
Why is it called `RouteValidator`? Looks like it validates path.

Reply:
it's the standard https://tools.ietf.org/html/rfc3986#section-3.3